### PR TITLE
Support iOS app spawn on unc0ver 6.x

### PR DESCRIPTION
**Pull request by mrmacete**
By skipping the call to substituted which eventually results in resuming the app too early, only for the PIDs of the apps spawn by Frida.

Needs more testing to find out possible edge cases of this approach, but to my initial tests it fixes the spawn issue on unc0ver 6.x

### CODE REVIEW

File: src/darwin/agent/launchd.js [hunk 1]
```diff
@@ -252,3 +271,44 @@ function findClosestMachHeader(address) {
...
   }
 }
+
+function findInserterResume() {
+  if (_cachedInseterImpl !== null) {
+    if (_cachedInseterImpl.isNull()) {
+      return null;
+    }
+    return _cachedInseterImpl;
+  }
```
Comment (oleavr) [inline src/darwin/agent/launchd.js:281]: I guess there's only a single caller so there's no need to optimize this?

**Review by oleavr — CHANGES_REQUESTED** [review]
w00t! 💥

File: src/darwin/agent/launchd.js (hunk 1 above)
Comment (mrmacete) [reply src/darwin/agent/launchd.js:281]: right!

File: src/darwin/agent/launchd.js [hunk 2]
```diff
@@ -252,3 +270,34 @@ function findClosestMachHeader(address) {
     cur = cur.sub(4096);
   }
 }
+
+function findInserterResume() {
+  const candidates = Process.enumerateModules().filter(x => x.name === 'substitute-inserter.dylib');
+  if (candidates.length !== 1) {
+    return null;
+  }
```
Comment (oleavr) [inline src/darwin/agent/launchd.js:278]:
```suggestion
  if (candidates.length !== 1)
    return null;
```

File: src/darwin/agent/launchd.js [hunk 3]
```diff
@@ -252,3 +270,34 @@ function findClosestMachHeader(address) {
...
+
+  const { base, size } = candidates[0];
+  const signature = '8? 00 00 b4 e0 03 00 91 ?? ?? 00 9? e4 0f 40 b9 e0 03 00 91 e1 07 00 32 82 05 80 52 83 05 80 52 05 00 80 52';
+
+  const matches = Memory.scanSync(base, size, signature);
+  if (matches.length !== 1) {
+    return null;
+  }
```
Comment (oleavr) [inline src/darwin/agent/launchd.js:286]:
```suggestion
  if (matches.length !== 1)
    return null;
```

File: src/darwin/agent/launchd.js [hunk 4]
```diff
@@ -252,3 +270,34 @@ function findClosestMachHeader(address) {
...
+
+  let cursor = matches[0].address.sub(4);
+  const end = cursor.sub(1024);
+  while (cursor.compare(end) >= 0) {
+    try {
+      const instr = Instruction.parse(cursor);
+      if (instr.mnemonic.startsWith('ret')) {
+        return cursor.add(4).sign();
+      }
```
Comment (oleavr) [inline src/darwin/agent/launchd.js:295]:
```suggestion
      if (instr.mnemonic.startsWith('ret'))
        return cursor.add(4).sign();
```

**Review by oleavr — CHANGES_REQUESTED** [review]

**Review by oleavr — APPROVED** [review]
💥
